API/hermes_abi/hermes_abi.h

```diff
@@ -0,0 +1,400 @@
+/// by get_property_names on a HostObject.
+struct HermesABIPropNameIDListVTable {
+  void (*release)(HermesABIPropNameIDList *);
+};
+struct HermesABIPropNameIDList {
+  const HermesABIPropNameIDListVTable *vtable;
+  const HermesABIPropNameID *props;
+  size_t size;
+};
+
+/// Define the structure for host objects. This is designed to recreate the
+/// functionality of jsi::HostObject.
+struct HermesABIHostObjectVTable {
+  HermesABIValueOrError (
+      *get)(HermesABIHostObject *, HermesABIContext *, HermesABIPropNameID);
+  HermesABIVoidOrError (*set)(
+      HermesABIHostObject *,
+      HermesABIContext *,
+      HermesABIPropNameID,
+      const HermesABIValue *);
+  HermesABIPropNameIDListPtrOrError (
+      *get_property_names)(HermesABIHostObject *, HermesABIContext *);
+  void (*release)(HermesABIHostObject *);
+};
+struct HermesABIHostObject {
+  const HermesABIHostObjectVTable *vtable;
+};
+
+struct HermesABINativeStateVTable {
+  void (*release)(HermesABINativeState *);
+};
+struct HermesABINativeState {
+  const HermesABINativeStateVTable *vtable;
+};
+
+struct HermesABIVTable {
+  /// Create a new instance of a Hermes Runtime, and return a pointer to its
+  /// associated context. The context must be released with
+  /// release_hermes_runtime when it is no longer needed.
+  HermesABIContext *(*make_hermes_runtime)();
+  /// Release the Hermes Runtime associated with the given context.
+  void (*release_hermes_runtime)(HermesABIContext *);
+
+  /// Methods for retrieving and clearing exceptions. An exception should be
+  /// retrieved if and only if some method returned an error value.
+  /// Get and clear the stored JS exception value. The returned value is
+  /// guaranteed to not be another exception. This should be called exactly once
+  /// after an exception is thrown.
+  HermesABIValue (*get_and_clear_js_error_value)(HermesABIContext *);
+  HermesABIASCIIRef (*get_native_exception_message)(HermesABIContext *);
+  /// Clear the stored native exception message. This should be called exactly
+  /// once after the message is retrieved.
+  void (*clear_native_exception_message)(HermesABIContext *);
+
+  /// Set the current error before returning control to the ABI. These are
+  /// intended to be used to throw exceptions from HostFunctions and
+  /// HostObjects.
+  void (*set_js_error_value)(HermesABIContext *, const HermesABIValue *);
+  void (
+      *set_native_exception_message)(HermesABIContext *, const char *, size_t);
+
+  HermesABIPropNameID (
+      *clone_prop_name_id)(HermesABIContext *, HermesABIPropNameID);
+  HermesABIString (*clone_string)(HermesABIContext *, HermesABIString);
+  HermesABISymbol (*clone_symbol)(HermesABIContext *, HermesABISymbol);
+  HermesABIObject (*clone_object)(HermesABIContext *, HermesABIObject);
+  HermesABIBigInt (*clone_big_int)(HermesABIContext *, HermesABIBigInt);
+
+  /// Check if the given buffer contains Hermes bytecode.
+  bool (*is_hermes_bytecode)(const uint8_t *, size_t);
```

Comment:
Should we add the `HermesABIContext *` parameter for consistency even if it is not used?

Reply:
I think not, this certainly isn't the only function in the vtable that doesn't take a `HermesABIContext`, and it seems unreasonable to require creating a context to check if something is bytecode (since that is currently a static function in our C++ API).